Leanplum-SDK/Classes/Internal/Leanplum.m

```diff
@@ -1948,7 +1935,25 @@ + (void)track:(NSString *)event withValue:(double)value andInfo:(NSString *)info
     if ([[UIApplication sharedApplication] applicationState] != UIApplicationStateActive) {
         arguments[@"allowOffline"] = @YES;
     }
+    return arguments;
+}
 
++ (void)track:(NSString *)event withValue:(double)value andInfo:(NSString *)info
+      andArgs:(NSDictionary *)args andParameters:(NSDictionary *)params
+{
+    RETURN_IF_NOOP;
+    LP_TRY
+    
+    // Track should not be called in background.
+    if (![NSThread isMainThread]) {
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            [self track:event withValue:value andInfo:info andArgs:args andParameters:params];
+        });
+        return;
```

Comment:
The problem is , we are not throwing any errors stating that "you have called track on main thread etc", it just returns. I know this is existing code, we might have to think on this together, cos of the way the callbacks are setup.